test/reconnect_test.exs

```diff
@@ -0,0 +1,44 @@
+defmodule RigInboundGatewayWeb.ReconnectTest do
+  use ExUnit.Case, async: true
+
+  require Logger
+
+  alias HTTPoison
+  alias Rig.Connection.Codec
+
+  @event_hub_http_port Confex.fetch_env!(:rig, RigInboundGatewayWeb.Endpoint)[:http][:port]
+
+  test "Initialize connection, then disconnect and reconnect" do
+    assert {:ok, client1} = SseClient.connect()
+    {event1, _} = SseClient.read_welcome_event(client1)
```

Comment:
the idea was to use the returned client for the next interactions... if you do it like this, any future updates to SseClient that make use of this will break this code in subtle ways

Reply:
What do you mean? 😅

Reply:
the returned tuple is `{event, client}`, where `event` is the event read from the stream and `client` is the _potentially updated_ client reference. So you could do `{event1, client1} = SseClient.read_welcome_event(client1)`